client/my-sites/themes/controller.jsx

```diff
@@ -96,6 +97,12 @@ export function fetchThemeFilters( context, next ) {
 	}
 
 	const unsubscribe = store.subscribe( () => {
+		// A notice is added to the state as the "on error" handler in the getThemeFilters action.
+		if ( getNotice( store.getState(), 'fetch-theme-filters-error' ) ) {
```

Comment:
It's a bummer and feels a bit fragile we have to peek into the notices to see if there was an error. I wonder if this is worth introducing another reducer subtree to the `themes` redux tree, called `themeFiltersError`, and it could be as simple as a boolean flag, then we would dispatch an action to make it `true` right where we call `reportRerror()` in the data layer. Sounds like it would be better and more reliable than betting on the notices state. That way we also won't have to create a new selector to fetch a particular notice. What do you think?

Reply:
I was also thinking about this. I think it would be useful to save the error message as well, so we can report it in the server logs.

That said, I don't think it's technically any more or less fragile -- the on error handler for the request currently dispatches an action which happens to set the notice. This different approach would still dispatch an action, it'd just set a different part of the state.

But I'll try it anyways, since it at least makes more sense!